hugr/src/hugr.rs

```diff
@@ -196,8 +196,10 @@ impl Hugr {
         extension_registry: &ExtensionRegistry,
     ) -> Result<(), ValidationError> {
         resolve_extension_ops(self, extension_registry)?;
+        self.validate_no_extensions(extension_registry)?;
         self.infer_extensions()?;
-        self.validate(extension_registry)?;
+        #[cfg(feature = "infer_extensions")]
```

Comment:
Should we put this around the line above too (in a `{...}` block)? And then drop the `#cfg` gating inside `Hugr.infer_extensions`? (So the latter does its stuff regardless of the feature, and we just avoid calling it unless the feature is on - or unless you *really* want to?)   Does that work?